### PR TITLE
Detect against invalid variant index for LibStdC++ std::variant data formatters

### PR DESCRIPTION
**Pull request by jeffreytan81**
https://github.com/llvm/llvm-project/pull/68012/files added new data formatters for LibStdC++ std::variant. 

However, this formatter can crash if std::variant's index field has invalid value (exceeds the number of template arguments). 
This can happen if the current IP stops at a place std::variant is not initialized yet. 

This patch fixes the crash by ensuring the index is a valid value.

### CODE REVIEW

**Review by clayborg — COMMENTED** [review]
We should add a test. Should be easy to access the "_M_index" child and set its value to something invalid and then make sure the summary is "<Invalid>".

**Comment by jeffreytan81** [comment]
> We should add a test. Should be easy to access the "_M_index" child and set its value to something invalid and then make sure the summary is "".

Sure, I can do that.

File: lldb/examples/synthetic/gnu_libstdcpp.py [hunk 1]
```diff
@@ -914,6 +914,11 @@ def get_variant_npos_value(index_byte_size):
     if index == npos_value:
         return " No Value"
 
+    # Invalid index can happen when the variant is not initialized yet.
+    template_arg_count = data_obj.GetType().GetNumberOfTemplateArguments()
+    if index >= template_arg_count:
+        return " <Invalid>"
+
     active_type = data_obj.GetType().GetTemplateArgumentType(index)
```
Comment (clayborg) [inline lldb/examples/synthetic/gnu_libstdcpp.py:922]: If this is what was crashing, we should modify the GetTemplateArgumentType(<index>) to not crash with an inalid index as part of this fix.

**Review by clayborg — CHANGES_REQUESTED** [review]

**Review by clayborg — APPROVED** [review]

**Comment by rupprecht** [comment]
This seems to cause `TestTemplatePackArgs.py` to fail, e.g. https://lab.llvm.org/buildbot/#/builders/68/builds/61819

```shell
  File "/home/worker/2.0.1/lldb-x86_64-debian/llvm-project/lldb/test/API/lang/cpp/class-template-parameter-pack/TestTemplatePackArgs.py", line 42, in test_template_argument_pack
    self.assertEqual(nested_template.GetTemplateArgumentType(2).GetName(), "bool")
AssertionError: '' != 'bool'
```

**Comment by DavidSpickett** [comment]
I've reverted it as the failure reason wasn't immediately obvious to me. Looks like it will reproduce quite easily.

**Pull request merged**